master/internal/config/resource_config.go

```diff
@@ -25,7 +25,8 @@ func (r *ResourceConfig) ResolveResource() error {
 	if r.ResourceManager.AgentRM == nil && r.ResourceManager.KubernetesRM == nil {
 		r.ResourceManager.AgentRM = &AgentResourceManagerConfig{}
 	}
-	if r.ResourceManager.AgentRM != nil && r.ResourcePools == nil {
+	if r.ResourcePools == nil &&
+		(r.ResourceManager.AgentRM != nil || r.ResourceManager.KubernetesRM != nil) {
```

Comment:
Good catch, I knew it would change the name, but I didn't realize that would have any effects. It does break just clicking the fork button, but you can manually open the full config in the UI and change the resource pool. Think this warrants any workarounds in the code, or can we just call it a release note?

Reply:
To fix it in the code we could make the default pool for k8s ```kubernetes``` which could be okay? 

Could we get away with a migration updating expconf where pool is ```kubernetes``` (would be pretty weird if someone had an agent pool named ```kubernetes```)?

Documenting could be fine too, but is a little unfortunate to cause an error with completely default master config upgrade